parser.go

```diff
@@ -47,8 +50,13 @@ type parser struct {
 }
 
 func newParser(r io.Reader, opts parserOptions) *parser {
+	size := opts.ReaderBufferSize
+	if size < minReaderBufferSize {
+		size = minReaderBufferSize
+	}
+
 	return &parser{
-		buf:     bufio.NewReader(r),
+		buf:     bufio.NewReaderSize(r, size),
```

Comment:
What is the rationale to use `bufio.NewReaderSize` rather than `bufio.NewReader`? What is not working with `bufio.NewReader`?

Reply:
I needed to read large values. However, bufio.NewReader() limits the internal buffer size to 4kB. 

I needed this size to be configurable, so I added a config option for it and passed it to NewReaderSize(). Hence the ‘large’ in the title of the pull request ;)